### PR TITLE
Firestore Sync Client Instrumentation

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Adds instrumentation for the following classes in Google Cloud Firestore's library.
  * Query
  * AggregationQuery
  * CollectionGroup
  * WriteBatch
  * BulkWriteBatch
  * Transaction

Depends On: #879

### CODE REVIEW

**Review by hmstepanek — CHANGES_REQUESTED** [review]
Looks really good! Just a couple small things.

File: newrelic/hooks/datastore_firestore.py [hunk 1]
```diff
@@ -12,22 +12,40 @@
...
-from newrelic.api.datastore_trace import wrap_datastore_trace
+from newrelic.api.datastore_trace import DatastoreTrace, wrap_datastore_trace
 from newrelic.api.function_trace import wrap_function_trace
 from newrelic.common.async_wrapper import generator_wrapper
-from newrelic.api.datastore_trace import DatastoreTrace
+from newrelic.common.object_wrapper import wrap_function_wrapper
+
+
+def _get_object_id(obj, *args, **kwargs):
+    try:
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/hooks/datastore_firestore.py:22]: Yeah that's probably better to collapse all 3

File: tests/datastore_firestore/test_collections.py [hunk 2]
```diff
@@ -12,25 +12,26 @@
...
+def _exercise_collections(collection):
     collection.document("DoesNotExist")
     collection.add({"capital": "Rome", "currency": "Euro", "language": "Italian"}, "Italy")
     collection.add({"capital": "Mexico City", "currency": "Peso", "language": "Spanish"}, "Mexico")
-    
+
     documents_get = collection.get()
     assert len(documents_get) == 2
-    documents_stream = list(collection.stream())
+    documents_stream = [_ for _ in collection.stream()]
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply tests/datastore_firestore/test_collections.py:32]: It wasn't consuming the iterator on older python versions, it just returned the generator object in a list. That might have been a change in behavior in Python 3.9

**Review by hmstepanek — APPROVED** [review]
Looks good!

**Pull request merged**